prometheus/metric.go

```diff
@@ -388,8 +395,43 @@ func (m metricBuckets) update(value float64) {
 	}
 }
 
-func ftoa(f float64) string {
-	return strconv.FormatFloat(f, 'g', -1, 64)
+// This function builds a string of column-separated float representations of
+// the given list of buckets, which is then split by calls to nextLe to generate
+// the values of the "le" label for each bucket of a histogram.
+//
+// The intent is to keep the number of dynamic memory allocations constant
+// instead of increasing linearly with the number of buckets.
+func le(buckets []float64) string {
+	if len(buckets) == 0 {
+		return ""
+	}
+
+	b := make([]byte, 0, 8*len(buckets))
+
+	for i, v := range buckets {
+		if i != 0 {
+			b = append(b, ':')
+		}
+		b = ftoa(b, v)
+	}
+
+	return *(*string)(unsafe.Pointer(&reflect.StringHeader{
+		Data: uintptr(unsafe.Pointer(&b[0])),
+		Len:  len(b),
+	}))
+}
+
+func nextLe(s string) (head string, tail string) {
+	if i := strings.IndexByte(s, ':'); i >= 0 {
+		head, tail = s[:i], s[i+1:]
+	} else {
+		head = s
+	}
+	return
+}
+
+func ftoa(b []byte, f float64) []byte {
```

Comment:
yeah but usually it returns a string with just the representation of the number. this append the representation to a already existing string.
if I use this ```go
str := ftoa([]byte("It would be nice..."), 42.42)
```
str would not be a valid representation of 42.42

maybe just a ```appendFloat```